app/models/host/discovered.rb

```diff
@@ -80,41 +80,44 @@ def attributes_to_import_from_facts
     super
   end
 
+  def detect_subnets
+    self.interfaces.each do |interface|
+      next unless interface.managed?
+      if interface.ip
+        interface.subnet = Subnet.subnet_for(interface.ip)
+        Rails.logger.info "Detected subnet #{subnet} for #{ip}" if subnet
+      end
+      if interface.ip6
+        interface.subnet6 = Subnet.subnet_for(interface.ip6)
+        Rails.logger.info "Detected subnet #{subnet6} for #{ip6}" if subnet6
+      end
+    end
+  end
+
   def populate_fields_from_facts(facts = self.facts_hash, ignored_type = nil)
     # detect interfaces and primary interface using extensions
     parser = super(facts, :foreman_discovery)
 
     # set additional discovery attributes
-    primary_ip = self.primary_interface.ip
-    unless primary_ip.nil?
-      subnet = Subnet.subnet_for(primary_ip)
-      if subnet
-        Rails.logger.info "Detected subnet: #{subnet} with taxonomy #{subnet.organizations.collect(&:name)}/#{subnet.locations.collect(&:name)}"
-      else
-        Rails.logger.warn "Subnet could not be detected for #{primary_ip}"
-      end
-      # set subnet
-      self.primary_interface.subnet = subnet
-      # set location and organization
-      if SETTINGS[:locations_enabled]
-        self.location = Location.find_by_title(Setting[:discovery_location]) ||
-          subnet.try(:locations).try(:first) ||
-          Location.first
-        Rails.logger.info "Assigned location: #{self.location}"
-      end
-      if SETTINGS[:organizations_enabled]
-        self.organization = Organization.find_by_title(Setting[:discovery_organization]) ||
-          subnet.try(:organizations).try(:first) ||
-          Organization.first
```

Comment:
Basically, this line was removed in the patch, it is not relevant I was thinking to remove it while touching this part. It does make little sense (sets more or less random organization).